lib/pf/Authentication/Source/HTTPSource.pm

```diff
@@ -0,0 +1,280 @@
+package pf::Authentication::Source::HTTPSource;
+
+=head1 NAME
+
+pf::Authentication::Source::HTTPSource
+
+=head1 DESCRIPTION
+
+=cut
+
+use pf::constants qw($TRUE $FALSE);
+use pf::Authentication::constants;
+use pf::Authentication::Action;
+use pf::Authentication::Source;
+use URI::Escape::XS qw(uri_escape uri_unescape);
+use WWW::Curl::Easy;
+use pf::log;
+use Readonly;
+use JSON;
+
+use Moose;
+extends 'pf::Authentication::Source';
+
+has '+type' => ( default => 'HTTP' );
+has 'protocol' => ( isa => 'Str', is => 'rw', default => 'http' );
+has 'host' => ( isa => 'Str', is => 'rw', default => '127.0.0.1' );
+has 'port' => ( isa => 'Int', is => 'rw', default => '10000' );
+has 'username' => ( isa => 'Maybe[Str]', is => 'rw', default => undef );
+has 'password' => ( isa => 'Maybe[Str]', is => 'rw', default => undef );
+has 'authentication_url' => ( isa => 'Str', is => 'rw', default => '' );
+has 'authorization_url' => ( isa => 'Str', is => 'rw', default => '' );
+
+# How and where to connect to the API
+Readonly our $API_PROTO => 'http';
+Readonly our $API_HOST => '127.0.0.1';
+Readonly our $API_PORT => '10000';
+
+# If your API uses HTTP basic authentication, set these to the username/password
+Readonly our $API_USERNAME => undef;
+Readonly our $API_PASSWORD => undef;
+
+# These are the two URLs necessary for this module to work
+# See the authenticate and match subroutines for the expected replies
+Readonly our $AUTHENTICATION_URL => '/authenticate.php';
+Readonly our $AUTHORIZATION_URL => '/authorize.php';
```

Comment:
Remove the use of the global variables API_USERNAME, API_PASSWORD, AUTHENTICATION_URL, AUTHORIZATION_URL